proto/v1/erc20.proto

```diff
@@ -2,63 +2,102 @@ syntax = "proto3";
 
 package erc20.types.v1;
 
-message BalanceChanges {
-  repeated BalanceChange balance_changes = 1;
+import "google/protobuf/timestamp.proto";
+
+message Events {
+  repeated Transfer transfers = 1;
+  repeated BalanceChange balance_changes = 2;
 }
 
 enum BalanceChangeType {
-  TYPE_UNKNOWN = 0; // cannot determine balance change
-  TYPE_1 = 1; // easy case where storage change is in the same call as the Transfer call
-  TYPE_2 = 2; // storage change is in a different call than the Transfer call
+  BALANCE_CHANGE_TYPE_UNSPECIFIED = 0; // cannot determine balance change
+  BALANCE_CHANGE_TYPE_1 = 1; // easy case where storage change is in the same call as the Transfer call
+  BALANCE_CHANGE_TYPE_2 = 2; // storage change is in a different call than the Transfer call
 }
 
 message BalanceChange {
-  string contract = 1;
-  string owner = 2;
-  string old_balance = 3;
-  string new_balance = 4;
+  // -- block --
+  uint64 block_num = 1;
+  string block_hash = 2;
+  google.protobuf.Timestamp timestamp = 3;
+  string date = 4;
 
-  string transaction = 5;
-  string storage_key = 6;
-  uint32 call_index = 7;
+  // -- transaction --
+  string transaction_id = 5;
 
-  string transfer_value = 8;
+  // -- call --
+  uint32 call_index = 6;
 
-  BalanceChangeType change_type = 9;
-}
+  // -- log --
+  uint32 log_index = 7; // Index is the index of the log relative to the transaction. This index is always populated regardless of the state revertion of the the call that emitted this log.
+  uint32 log_block_index = 8; // BlockIndex represents the index of the log relative to the Block.
+  uint64 log_ordinal = 9; // the block's global ordinal when the transfer was recorded.
 
-message ValidBalanceChanges {
-  repeated ValidBalanceChange valid_balance_changes = 1;
-}
+  // -- storage change --
+  string storage_key = 10;
+  uint64 storage_ordinal = 11;
+  string storage_address = 12;
 
-message ValidBalanceChange {
-  string contract = 1;
-  string owner = 2;
-  string old_balance = 3;
-  string new_balance = 4;
+  // -- balance change --
+  string contract = 20;
+  string owner = 21;
+  string old_balance = 22;
+  string new_balance = 23;
+  string amount = 24; // delta between old and new balance
 
-  string transaction = 5;
-}
+  // -- transfer --
+  string from = 25;
+  string to = 26;
+  string value = 27;
 
-message UnknownBalanceChanges {
-  repeated UnknownBalanceChange unknown_balance_changes = 1;
-}
+  // -- indexing --
+  uint64 version = 30; // latest version of the balance change (block_num << 32 + storage_ordinal)
```

Comment:
I don't like the `version` name, I feel it's really miss leading. `global_sequence` maybe.

Reply:
Also, the usage of `storage_ordinal` as an identifier raises some fear from my side. Ordinals are good for ordering but as soon as there is a small piece of indeterminism in the tracing, all the ordinals are shifted. 

If the ordinals are used only for ordering, in 99% of the case, the end processing will result in the same output even if the ordinals are different. But if they are used as IDs, it contaminates the output.

Could we use the relative ordering of the storage changes instead? That would lead to output differences only if ordering of storage balance changes would be different.

I'll try next week to check further the code and try to provide an alternative.

Reply:
Ah thanks for the insight! 👌  ok i'll implement some kind of relative `index` that doesn't use `storage_ordinal`

I didn't know it had some small piece of indeterminism, good to know!

Reply:
`global_sequence` does not exist on EVM? Don't see it for `storage_changes`, only `ordinal` is the only indexing field available

<img width="714" alt="image" src="https://github.com/user-attachments/assets/260c62aa-2220-4a0b-9dc1-3e03d6fbad82" />